### PR TITLE
Remove assignment to unused var, make other inventory changes for mypy (#75)

### PR DESCRIPTION
**Pull request by ssciolla**
This PR aims to satisfy a Codacy style warning about an unused variable by removing the assignment. In addition, some minor changes to code and type hints were made to fix `mypy` errors in `inventory.py`. The PR aims to resolve issue #75.

### CODE REVIEW

File: course_inventory/inventory.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def make_request_using_api_utils(url: str, params: Dict[str, Union[str, int]] =
     return response
 
 
-def slim_down_course_data(course_data: Sequence[Dict]) -> Sequence[Dict]:
+def slim_down_course_data(course_data: List[Dict]) -> List[Dict]:
```
Comment (lsloan) [inline course_inventory/inventory.py:67]: Why the change from `Sequence` → `List`?

File: course_inventory/inventory.py [hunk 2]
```diff
@@ -134,13 +134,13 @@ def gather_course_data_from_api(account_id: int, term_id: int) -> pd.DataFrame:
 
 # Function(s) - UDW
 
-def process_sis_id(orig_sis_id: str) -> Union[int, None]:
+def process_sis_id(orig_sis_id: str) -> Optional[int]:
```
Comment (lsloan) [inline course_inventory/inventory.py:137]: I considered using `Optional` in some of my code, but it made it look like a result value from the function was, well… optional.  I preferred `Union[…, None]` because it made it explicit that my function would return either the first type or `None`.

Comment (ssciolla) [reply course_inventory/inventory.py:137]: I can try switching back to that. I like `Union`, too.

File: course_inventory/inventory.py (hunk 1 above)
Comment (ssciolla) [reply course_inventory/inventory.py:67]: I was getting an error from `mypy` about argument types not matching, and this fixed it. I will recreate the error and post here.

Comment (ssciolla) [reply course_inventory/inventory.py:67]: ```course_inventory/inventory.py:113: error: Unsupported left operand type for + ("Sequence[Dict[Any, Any]]")```

Comment (ssciolla) [reply course_inventory/inventory.py:67]: I guess where I'm using `+` between lists on 113, it doesn't like that the type hint is `Sequence`?

File: course_inventory/inventory.py (hunk 2 above)
Comment (ssciolla) [reply course_inventory/inventory.py]: Done, I made the switch back here.

File: course_inventory/inventory.py (hunk 1 above)
Comment (jonespm) [reply course_inventory/inventory.py:67]: From the readme. "Dict and List are aliases for the built-ins dict and list, respectively."

"Iterable, Sequence, and Mapping are generic types that correspond to Python protocols. For example, a str object or a List[str] object is valid when Iterable[str] or Sequence[str] is expected."

So it seems like the way Python duck typing works it will throw an error for these operands if it's not always possible to do the check. You can probably also check for isinstance(var, dict) and have mypy be good with it too later in the code. What you have here seems correct tho. Still learning this stuff.
